---
title: 'Lab 4: Advanced bioinformatics'
author: "Henna Patel and Persis Adonteng"
date: "10/2/2020"
output: html_document
---

<h2>
<span style="color: red;"> 77/100. Use the Markdown notations for code, italics, etc, please. Also, ask me questions when you dont understand what the questions are in the lab. The query question took 10 pts out of your final grade...</span>
</h2>

# Advanced bioinformatics

## Notes:

-   Session 1 will be done on Oct. 16
-   Session 2 will be done in Oct. 23
-   Lab is due Oct. 30

***

# Introduction

This lab will be focused in understanding advanced bioinformatics.

**Section 1** wil focus on how to identify open reading frames and a gene based on searches on the ORF finder, translate the largest ORF and navigate through NCBI as we did on the theory class

**Section 2** will focus on how to use the cluster to create a BLAST and HMMER database of genes of interest.

***

# Section 1: Open reading frames, translating sequences and BLAST searches to identify the role and functionality of a mystery gene


1. Download the `unknown_seq.fasta` file from the Moodle and open it on `atom`. This is a nucelotide strand from a segment of a chromosome of a cool species that we will try to identify.

2. Go to [NCBI's ORF Finder](https://www.ncbi.nlm.nih.gov/orffinder/) and upload your sequence by copying and pasting it. Keep the parameters by default and submit the form. Answer the following questions:

***

> Questions:

> 1. How many ORF do you find in the sequence?

Answer: There are 5 ORF in the sequence. 

> 2. Which ORF is the longest, how long it is (Both in NT and AA), and in which strand (Plus or minus) is located?

Answer: ORF 1 is the longest. It has 153 NT and 50 AA. It is in + strand. 

> 3. In what frame does the longest ORF is in? What does this mean?

Answer: The longest ORF is in first frame. ORF 1 is the part of the first open reading frame. 

> 4. Click on Display ORF and copy and paste the Protein sequence and the nucleotide sequence of the longest ORF here

Answer: Protein sequence: 
>lcl|ORF1
MVMITLDRYNVIVRGMAAAPLTHKKATLLLLFVWIWSGGWTILPFFGWSR

Nucleotide sequence:
>lcl|ORF1 CDS

ATGGTGATGATAACACTAGACCGCTACAACGTGATTGTTAGAGGTATGGC
AGCTGCACCTTTGACCCACAAGAAAGCCACTCTGCTCCTCCTGTTTGTCT
GGATCTGGTCAGGTGGTTGGACTATCCTGCCTTTCTTCGGATGGAGCAGG
TAG

> 5. Is there any ORF on the backwards direction? (-)

Answer: ORF 5 is on the negative strand. 

***

3. Now, take your original sequence and go to [NCBI BLAST ](https://blast.ncbi.nlm.nih.gov/Blast.cgi). Answer the following question:


***

> 6. This is DNA data, so what kinds of BLAST's can you use? 

Answer: The nucleotide blast can be used to view this DNA data.

***

4. First, lets do a nucleotide BLAST. Click on nucleotide BLAST, and copy and paste your original sequence. Now, select the **Nucleotide collection** database and run a megablast. Answer the following questions:


***

> 7. What are the best 3 hits. Why do you consioder them the best hits?

Answer: The first three sequences are best hits because max score is higher, E value closer to 0 and the percentage is closer to 100%. They have the highest query cover percentage of bases/AA of the query that return a hit. 

> 8. Include the accesion numbers of the best three hits.

Answer: The accession numbers are AY190508.1, AY190514.2 and AY190512.2. 

***

5. Go to the Graphic Summary of the BLAST results. Take a screenshot and add it below and answer the following questions

***

> 9. Add screenshot here
\![Graphic summary of the BLAST results](sequences/Screen Shot 2020-10-16 at 3.34.31 PM.png)

> 10. Does this image support your results from question 7? Explain why briefly.

Answer: The first three horizontal bars (red) have the alignment scores of >= 200 representing the best hit. There are no breaks between the horizontal bars which means that almost the entire query score is covered.

<span style="color: red;"> But how do they support (or not) your results from question 7? (-3 pt) </span>

***

6. Go to the Taxonomy window of the BLAST results. Answer the following question:

***

> 11. What species has the highest score? What about the other species? Are they closely related to each other?

Answer: Limulus polyphemus has the highest score (1406).  The last four species are closely related as they are classified as bees. However, Macropis nuda and Macropis europaea are more closely related as they have the score of 82.4 and 80.5 respectively.

<span style="color: red;"> Scientific notation for species! Use italics for binomial names (-1 pt) </span>

***

7. After doing the BLAST, click on the hit with the best score. You should find the accession number there (AY190508.1). Click on it and answer the following question:

***

> 12. What kind of is this? A FASTA or a GenBank

Answer: It is a GenBank file. 

> 13. Copy and paste the protein region coded by the CDS

Answer: 
"RYNVIVRGMAAAPLTHKKATLLLLFVWIWSGGWTILPFFGWSRY
                     VPEGNLTSCTVDYLTKDWSSASYVIIYGLAVYFLPLITMIYCYFFIVHAVAEHEKQLR
                     EQAKKMNVASLRANADQQKQSAECRLAKVAMMTVGLWFMAWTPYLIIAWAGVFSSGTR
                     LTPLATIWGSVFAKANSCYNPIVYGISHPRYKAALYQRFPSLACGSGESGSDVKSEAS
                     ATTTMEEKPKSPEA"

***

8. On the panel on the right, you'll find a link to the *More about the gene LOC106470356*. This means we have a whole gene associated to our query gene!

Answer the following questions:

***

> 14. What's the name of the gene?

Answer: lateral eye opsin

> 15. Check the organism's name. Is this what you found at the beginning of the lab?

Answer: Yes, It is the same name (Limulus polyphemus). 

<span style="color: red;"> Scientific notation for species! Use italics for binomial names (-1 pt) </span>

> 16. Go to the genomic regions? How many exons does the gene has? How did you come to this conclusion?

Answer: The gene has two exons. In the genomic region, Click on one of the exon and it lists on the square box "exon 2 of 2" which help us to come to this conclusion. 

***

10. Go down to the NCBI Reference Sequences (RefSeq) - mRNA and Proteins section. 

11. Click on the UniProtKB/Swiss-Prot link. Answer the following question:

***

> 17. What database are we in?

Answer: The databse is UniProt. 

> 18. Is the organism still the same?

Answer: The name of the organism is still the same (Limulus polyphemus). 
<span style="color: red;"> Scientific notation for species! Use italics for binomial names (-1 pt) </span>

> 19. What is the proposed function for this gene?

Answer: The gene consists of apoprotein opsin that are covalently linked to the cis retinal. They use visual pigment that meditate vision. 

> 20. What are the molecular functions and biological processes that this gene is involved with?

Answer: The molecular functions are G protein-coupled receptor activity, Photoreceptor protein, Receptor, Retinal protein, and Transducer. The biological processes are phototransduction, protein-chromophore linkage, visual perception and sensory transduction. 

***


12. Extract the FASTA protein sequence from the LOC106470356 gene (You can find it on the RefSeq submenu, on the mRNA and Protein segment, click on the NP_001301089.1 link). Then, go to [Pfam](http://pfam.xfam.org/) and search the sequence you just copied on the Sequence Search window. Answer the following questions based on the **Significant Pfam matches**:

***

> 21. How many protein domains can we find in this sequence?

Answer: There is one protein domain named 7tm_1 in this sequence. 

> 22. What domain is it?

Answer: The domain is 7tm_1. 

> 23. Is this a good E-value to determine a good hit?

Answer: The E value 2.2e-53 is good to determine a good hit as it is closer to 0. 

***

13. Click on the domain name. Answer these questions:

***

> 21. How many architectures of this domain are found?

Answer: This domain has 735 architectures.

> 22. How many sequences are found that contain this domain? 

Answer: There are 84481 sequences found in this domain.

***

14. Click on the the Species link on the menu on the left. Answer these questions:

***

> 23. What are the kindom, phylum and class with more sequences that contain this domain?

Answer: The kingdom is Metazoa. The phylum is Chordata, Arthropoda, Nematoda and uncatategorized eukaryotes. The class are Actinopteri, Insecta, Mamalia, Aves, uncategorized eukaryotes and Chromadorea.  

***

## Homework 1: 

Based on all the information that we found today, write a summary of the gene and all the information we were able to obtain from the search. Include gene ID, accession numbers for mRNA and protein sequence. Two paragraphs are more than enough.

***

> Answer here: 

The first three sequences from the nucleotide BLAST was selected based on the higher max score, and e values closer to 0. The first sequence with LOCI106470356 was used to analyze the gene. The accession numbers for 3 best hits are AY190508.1, AY190514.2 and AY190512.2 The accession numbers for the gene studied is AY190508.1. The gene is lateral eye opsin in the species Limulus polyphemus. The gene consists of apoprotein opsin that are covalently linked to the cis retinal. They use visual pigment that meditate vision. The molecular function this gene is involved in are G protein-coupled receptor activity, photoreceptor protein, retinal protein and other functions. The biological processes are phototransduction, sensory transduction, visual perception and many other functions. The gene has two exons and one protein domain (7tm_1). The kingdom that contains this domain is Metazoa. The phyla is Chordata, Arthropoda, Nematoda and uncatategorized eukaryotes which contains this domain. The class are Actinopteri, Insecta, Mamalia, Aves, uncategorized eukaryotes and Chromadorea which contains this domain.

<span style="color: red;"> Very good, but please use scientific notation for species! Use italics for binomial names (-1 pt) </span>

***


# Section 2: Creating our own searches using BLAST and HMMER in the cluster.

## Creating databases using BLAST

Today we are going to learn how to create our own databases in the cluster based on a list of sequences and search to identify query sequences from species of interest.

In our case, we are going to use the database from *Phytophthora-ID*, a database I constructed a while ago to detect species of *Phytophthora* based on the ITS molecular marker.

1. Connect to the cluster and create a new folder inside your MBB101 folder called `Lab_4`
2. Create a second folder inside of `Lab_4` called blastDB and go into it.
3. Download the sequences from this link `http://phytophthora-id.org/files/Phytophthora-ID-ITS-version-1.17.fasta` into the `Lab_4/blastDB` folder
4. We are going to use the `unknown_sequences.fasta` file as query for the unknown files

Answer the following questions:

***

> 1. How many sequences are in the `unknown_sequences.fasta` and the `Phytophthora-ID-ITS-version-1.17.fasta` files? Add the code you used to check the sequences.

Answer: There are 2 sequences in the unknown_sequences.fasta and 270 sequences in the Phytophthora-ID-ITS-version-1.17.fasta file. 

The following commands were used:

grep ">" unknown_sequences.fasta | wc -l

grep ">" Phytophthora-ID-ITS-version-1.17.fasta | wc -l

<span style="color: red;"> Use code notation for code!(-1 pt) </span>

> 2. Are these files nucleotide or protein sequences? Add the code you added to check the sequences.

Both files are nucleotide sequences because they contain nucleotide bases. We created new files for each Fasta file without the header and checked to see if there are any protein seqences present. There were no protein sequences present. 

unknown_sequences.fasta 

1. sed -n '1,2 p' unknown_sequences.fasta > Query_1
1. grep -v '>' Query_1 > Query_sequence
1. grep -o M Query_sequence | wc -l


 Phytophthora-ID-ITS-version-1.17.fasta
 
1. sed -n '1,2 p' Phytophthora-ID-ITS-version-1.17.fasta > Phy_1
1. grep -v '>' Phy_1 > Phy_1.1
1. grep -o M Phy_1.1 | wc -l

<span style="color: red;"> Great job, but you need to use code blocks for code... (-2 pt) </span>

***

Now, we are going to create the BLAST database. To create this we will use the following template of code:

```
makeblastdb -in list_of_reference_seq -dbtype type_of_molecule -out database_name
```

Answer the following question:

***

> 3. Modify the template code by changing the `list_of_reference_seq`, `type_of_molecule` and `database_name` and add it below.
makeblastdb -in unknown_sequences.fasta -dbtype nucl -out ITS

makeblastdb -in Phytophthora-ID-ITS-version-1.17.fasta -dbtype nucl -out ITS2

<span style="color: red;"> Use code notation for code!(-1 pt) </span>

***

5. Run the code after modification and make sure it worked by doing an `ls` and checking that the `database_name.nhr`, `database_name.nin`, `database_name.nsq` files are there.

Yes, all three files were there. 

 ITS2.nhr
 
 ITS2.nin
 
 ITS2.nsq

6. Now, we are going to run a BLAST search to identify the unknown sequences in the `unknown_sequences.fasta`

The following is template code to run BLAST:

```
blast_program -query query_file -db database_name -out output_file -evalue 1e-10 -outfmt 6 -max_target_seqs 7
```

Answer the following questions

***

> 4. Modify the template code by changing the `blast_program` to the blast program we should be using (blastp, blastn, tblastn, tblastx). Explain your answer briefly.

Answer: We used blastn because we have a nucleotide sequences that need to be searched through the database using a query file. 

> 5. Complete the template code by modifying the `query_file` and `database_name` and add it below.

Answer: blastn -query unknown_sequences.fasta -db ITS2 -out Phy_q1 -evalue 1e-10 -outfmt 6 -max_target_seqs 7

<span style="color: red;"> Use code notation for code!(-1 pt) </span>

> 6. Explain briefly what the `-evalue 1e-10`, `-outfmt 6`, `-max_target_seqs 7` and flags mean.

Answer: `-evalue 1e-10`: means that we want a evalue close to zero as the closer it is to zero, the match is more significant.
`-outfmt 6`: gives us a output format table, amd `-max_target_seqs 7`: means that we want only 7 target sequence from both query 1 and query 2. Which gave us a total of 14 different sequences. 

\![query table](sequences/noww.png)


***

7. Run the modified code. Check that you have some results by making sure that your `output_file` has content on it.

8. In the following table, add the top 3 hits for each of the query sequences.

| qseqid  | sseqid | pident | length | mismatch | gapopen | qstart | qend | sstart | send | evalue | bitscore |
|---------|--------|--------|--------|----------|---------|--------|------|--------|------|--------|----------|
| Query_1 |AY369370.1 | 100.000 | 797|  0       |   0     |    1   |  797 |    1   |   797|     0.0| 1472     |
| Query_1 |AY369375.1|99.875|    797 |1          |      0   |     1   |   797   |   1 |   797|  0.0|1467     |
| Query_1 |AF266804| 95.119| 799|    35      |     4    |    1    |    797  |      1  |     797 |    0.0    |   1256       |

| qseqid  | sseqid | pident | length | mismatch | gapopen | qstart | qend | sstart | send | evalue | bitscore |
|---------|--------|--------|--------|----------|---------|--------|------|--------|------|--------|----------|
| Query_2 |L41373.1   |  100.000  |   798     |      0    |       0  |   1     |     798 |    1    |   798   |  0.0      |       1474   |
| Query_2 |AF271221 |   99.373     |    798    |     4     |  1       |      1  |   798   |    1  |   797   |      0.0  |         1445 |
| Query_2 | HQ261661     |   98.116     |    796    |     12     |     3    |     1   |     796 |    1    |    793  |      0.0  |    1384      |

Answer these following questions:

***

> 7. Based on your results, what are the most likely identities of each query? Explain why.

Answer:
Query_1 AY369370.1  Phytophthora hibernalis									
Query_1 AY369375.1   Phytophthora hibernalis									
Query_1 AF266804     Phytophthora lateralis

Query_2 L41373.1   Phytophthora macrochlamydospora								
Query_2 AF271221   Phytophthora richardiae									
Query_2 HQ261661   Phytophthora quininea 

These are the most likely identities of each query because they are the ones that is close to the hit on sequences in the Phytophthora-ID-ITS-version-1.17.fasta as they have e-value close to zero.

\![species names](sequences/names.png)

> 8. Within the `Phytophthora-ID-ITS-version-1.17.fasta` you can extract the names of each species in the database. What are the species of each of the most likly sequences? 

(We weren't sure if you wanted just the 14 hits from the query or all. So we provided all 270 species )

>AB217670.1 Phytophthora capsici 

>AB217671.1 Phytophthora cactorum 

>AB217675.1 Phytophthora cinnamomi 

>AB217682.1 Phytophthora nicotianae 

>AB217684.1 Phytophthora sojae 

>AB217687.1 Phytophthora vignae 

>AF139367.1 Phytophthora alni subsp. uniformis 

>AF139369.1 Phytophthora cambivora 

>AF266761 Phytophthora rubi 

>AF266762 Phytophthora fragariae 

>AF266763 Phytophthora cambivora 

>AF266764 Phytophthora cinnamomi 

>AF266765 Phytophthora cajani 

>AF266766 Phytophthora vignae 

>AF266767 Phytophthora melonis 

>AF266768 Phytophthora sinensis 

>AF266769 Phytophthora sojae 

>AF266770 Phytophthora heveae 

>AF266771 Phytophthora katsurae 

>AF266772 Phytophthora cactorum 

>AF266773 Phytophthora idaei 

>AF266774 Phytophthora pseudotsugae 

>AF266775 Phytophthora tentaculata 

>AF266776 Phytophthora nicotianae 

>AF266777 Phytophthora mirabilis 

>AF266778 Phytophthora phaseoli 

>AF266779 Phytophthora infestans 

>AF266782 Phytophthora megakarya 

>AF266784 Phytophthora botryosa 

>AF266785 Phytophthora citrophthora 

>AF266786 Phytophthora colocasiae 

>AF266787 Phytophthora capsici 

>AF266789 Phytophthora plurivora 

>AF266790 Phytophthora multivesiculata 

>AF266792 Phytophthora humicola 

>AF266794 Phytophthora megasperma 

>AF266795 Phytophthora taxon asparagi 

>AF266796 Phytophthora cryptogea 

>AF266797 Phytophthora erythroseptica 

>AF266798 Phytophthora drechsleri 

>AF266799 Phytophthora medicaginis 

>AF266800 Phytophthora trifolii 

>AF266802 Phytophthora primulae 

>AF266803 Phytophthora syringae 

>AF266804 Phytophthora lateralis 

>AF271221 Phytophthora richardiae 

>AF271222 Phytophthora insolita 

>AF279128 Phytophthora glovera 

>AF380147.1 Phytophthora brassicae 

>AF380150.1 Phytophthora porri 

>AF403506.1 Phytophthora pistaciae 

>AF408625.2 Phytophthora bisheria 

>AF449491.1 Phytophthora europaea 

>AF449494.1 Phytophthora psychrophila 

>AF449495.1 Phytophthora uliginosa 

>AF541887.1 Phytophthora gonapodyides 

>AF541888.1 Phytophthora gonapodyides 

>AF541891.1 Phytophthora gonapodyides 

>AF541892.1 Phytophthora gonapodyides 

>AF541893.1 Phytophthora megasperma 

>AF541895.1 Phytophthora megasperma 

>AF541896.1 Phytophthora megasperma 

>AF541898.1 Phytophthora megasperma 

>AF541900.1 Phytophthora taxon pgchlamydo 

>AF541901.1 Phytophthora taxon pgchlamydo 

>AF541903.1 Phytophthora gregata 

>AF541904.1 Phytophthora gregata 

>AF541905.1 Phytophthora taxon raspberry 

>AF541908.1 Phytophthora taxon forestsoil 

>AF541910.1 Phytophthora taxon walnut 

>AF541911.1 Phytophthora rosacearum 

>AF541912.1 Phytophthora inundata 

>AJ131986 Phytophthora quercina 

>AJ131987 Phytophthora iranica 

>AJ131989 Phytophthora clandestina 

>AJ131990 Phytophthora ilicis 

>AY208125.1 Phytophthora tropicalis 

>AY230190.1 Phytophthora pseudosyringae 

>AY242980.1 Phytophthora pseudosyringae 

>AY332654.1 Phytophthora nemorosa 

>AY369368.1 Phytophthora ramorum 

>AY369370.1 Phytophthora hibernalis 

>AY659736.1 Phytophthora parsiana 

>AY659737.1 Phytophthora parsiana 

>AY659738.1 Phytophthora parsiana 

>AY659739.1 Phytophthora parsiana 

>AY659740.1 Phytophthora parsiana 

>AY659741.1 Phytophthora parsiana 

>AY689131.1 Phytophthora alni subsp. alni 

>AY689134.1 Phytophthora alni subsp. uniformis 

>AY689136.1 Phytophthora alni subsp. multiformis 

>AY707987.1 Phytophthora hedraiandra 

>AY770742.1 Phytophthora ipomoeae 

>AY787025.1 Phytophthora gonapodyides 

>AY787028.1 Phytophthora cambivora 

>AY787029.1 Phytophthora cambivora 

>AY787030.1 Phytophthora cambivora 

>AY787031.1 Phytophthora cambivora 

>AY787032.1 Phytophthora syringae 

>AY787033.1 Phytophthora syringae 

>AY787034.1 Phytophthora syringae 

>AY940661 Phytophthora kernoviae 

>DQ286726 Phytophthora gallica 

>DQ297392.1 Phytophthora fallax 

>DQ297398.1 Phytophthora fallax 

>DQ297399.1 Phytophthora captiosa 

>DQ302411 Phytophthora bisheria 

>DQ313223 Phytophthora quercetorum 

>DQ396409.1 Phytophthora polonica 

>DQ396417.1 Phytophthora polonica 

>DQ396422.1 Phytophthora polonica 

>DQ988207.1 Phytophthora boehmeriae 

>DQ995184.1 Phytophthora austrocedrae 

>EF120469 Phytophthora foliorum 

>EF490682.1 Phytophthora siskiyouensis 

>EU583793.1 Phytophthora hydropathica 

>EU595771.1 Phytophthora irrigata 

>L41373.1 Phytophthora macrochlamydospora 

>HQ261661 Phytophthora quininea 

>EU301169.2 Phytophthora sp.personii 

>EU925375 Phytophthora sansomeana 

>EF564803 Phytophthora taxon cuyabensis 

>EF590256 Phytophthora taxon lagoariana 

>AY369375.1 Phytophthora hibernalis 

>AY251649.1 Phytophthora meadii 

>DQ988177.1 Phytophthora frigida 


>DQ988196.1 Phytophthora alticola 

>EU725805.1 Phytophthora pinifolia 

>EF590257.1 Phytophthora taxon sulawesiensis 

>EU035774.1 Phytophthora taxon novaeguinee 

>EF437222.1 Phytophthora taxon kunnunara 

>EF680320.1 Phytophthora taxon napoensis 

>EU164426.1 Phytophthora taxon zentmyerii 

>AY770741.1 Phytophthora andina 

>EU748545.1 Phytophthora mengei

>FJ469147.1 Phytophthora morindae

>FJ665225.1 Phytophthora plurivora


>FJ665227.1 Phytophthora plurivora

>FJ237521.2 Phytophthora multivora

>AB511826.1 Phytophthora chrysanthemi 

>AB437136.1 Phytophthora chrysanthemi 

>GQ847755.1 Phytophthora elongata 

>GU191219.1 Phytophthora capensis 


>HQ012948.1 Phytophthora litoralis 

>HQ012947.1 Phytophthora litoralis 

>HQ012946.1 Phytophthora litoralis 

>EU301173.2 Phytophthora gregata 

>EU301171.2 Phytophthora gregata 

>HQ012938.1 Phytophthora gregata 


>HQ012942.1 Phytophthora gregata 

>HQ012936.1 Phytophthora gibbosa 

>HQ012933.1 Phytophthora gibbosa 

>EU301155.2 Phytophthora thermophila 

>HQ012950.1 Phytophthora thermophila 

>EU301158.2 Phytophthora thermophila 

>HM752784.1 Phytophthora himalsilva 

>GQ324989.1 Phytophthora pini 

>AF279127.2 Phytophthora glovera 

>EU725806.1 Phytophthora pinifolia 

>HQ013215.1 Phytophthora arenaria 

>HQ013219.1 Phytophthora arenaria 

>HQ013225.1 Phytophthora constricta 

>EU301143.1 Phytophthora constricta 

>FJ217680.1 Phytophthora gemini 

>FJ217679.1 Phytophthora gemini 

>HQ917909 Phytophthora obscura 

>HQ917911 Phytophthora obscura 

>HQ917913 Phytophthora syringae 

>HQ917914 Phytophthora syringae 

>FJ237515.2 Phytophthora multivora 

>GQ303149.1 Phytophthora citricola III

>GU119914 Phytophthora niederhauserii 

>FJ802007 Phytophthora cinncmomi var. parvispora 

>GQ388261.1 Phytophthora cinnamomi var. parvispora 

>JF701435.1 Phytophthora fluvialis 

>JF701436.1 Phytophthora fluvialis 

>EU593261.2 Phytophthora fluvialis 

>HQ012953.1 Phytophthora taxon paludosa 

>JQ256377.1 Phytophthora bilorbang 

>JQ814693.1 Phytophthora litchii 

>JQ626594.1 Phytophthora riparia

>JQ626602.1 Phytophthora riparia

>JQ626600.1 Phytophthora riparia

>JQ626598.1 Phytophthora borealis

>GU191210.1 Phytophthora cinnamomi

>JX276017.1 Phytophthora polonica 

>HQ261646.1 Phytophthora polonica

>HQ261647.1 Phytophthora polonica 

>AF266793.2 Phytophthora lacustris

>HQ261507.1 Phytophthora botryosa 

>HQ261508.1 Phytophthora botryosa

>HQ261509.1 Phytophthora botryosa

>KC529657.1 Phytophthora pluvialis 

>HQ643205.1 Phytophthora citrophthora 

>HQ643207.1 Phytophthora citrophthora 

>HQ643208.1 Phytophthora citrophthora 

>AY659426.1 Phytophthora cryptogea

>AY659450.1 Phytophthora cryptogea

>AY659428.1 Phytophthora cryptogea

>AY659421.1 Phytophthora cryptogea

>AY659431.1 Phytophthora cryptogea

>AY659439.1 Phytophthora cryptogea

>AY659422.1 Phytophthora cryptogea

>AY659434.1 Phytophthora cryptogea

>AY659446.1 Phytophthora drechsleri

>AY659435.1 Phytophthora drechsleri

>AY659457.1 Phytophthora drechsleri

>AY659458.1 Phytophthora drechsleri

>AY659464.1 Phytophthora drechsleri

>AY659449.1 Phytophthora erythroseptica

>AY659445.1 Phytophthora erythroseptica

>EU240038.1 Phytophthora taxon oaksoil

>KC855330.1 Phytophthora pachypleura

>KF112853.1 Phytophthora mississippiae

>KF112851.1 Phytophthora mississippiae

>HQ261638.1 Phytophthora parsiana

>HQ261639.1 Phytophthora parsiana 

>HQ261637.1 Phytophthora parsiana 

>FJ666127.1 Phytophthora aquimorbida

>FJ666126.1 Phytophthora taxon aquatilis

>KC478759.1 Phytophthora taxon castitis

>KC478761.1 Phytophthora dauci

>KC478768.1 Phytophthora lactucae

>KC478773.1 Phytophthora cichorii

>JX951285.1 Phytophthora acerina

>KF192695.1 Phytophthora stricta

>KF192701.1 Phytophthora macilentosa

>JQ029956.1 Phytophthora amnicola

>JQ029958.1 Phytophthora amnicola

>HQ894399.1 Phytophthora pisi

>HQ894391.1 Phytophthora pisi

>HQ894394.1 Phytophthora pisi

>HQ894395.1 Phytophthora pisi

>HQ261713.1 Phytophthora sp.thermophilum

>HQ261712.1 Phytophthora sp.thermophilum

>HQ261620.1 Phytophthora mexicana

>HQ261710.1 Phytophthora sp.ohioensis

>KC295542.1 Phytophthora virginiana

>KC249960.1 Phytophthora hydrogena

>HQ643189.1 Phytophthora cinnamomi

>HQ261526.1 Phytophthora cinnamomi

>HQ643193.1 Phytophthora cinnamomi

>AB305065.1 Phytophthora fragaefolia

>AB688355.1 Phytophthora nagaii

>JX978155.1 Phytophthora occultans

>JX978167.1 Phytophthora terminalis

>HQ261491.1 Phytophthora aff. infestans

>HQ261588.1 Phytophthora infestans

>GU191228.1 Phytophthora sp. emzansi/CCB-2010 

>FJ560913.1 Phytophthora citricola sensu stricto

>AF266788.1 Phytophthora citricola E 

>AJ007370.1 Phytophthora plurivora

>AB688422.1 Phytophthora asiatica (cinnamomi var. robiniae)

>AB538396.1 Phytophthora asiatica (cinnamomi var. robiniae)

>KC478663.1 Phytophthora cinnamomi 

>KP295324.1 Phytophthora castaneae

>KP295323.1 Phytophthora castaneae

>KP295314.1 Phytophthora agathidicida

>KP295306.1 Phytophthora cocois

>AB894385.1 Phytophthora pseudolactucae

>AB894384.1 Phytophthora pseudolactucae

>KP295301.1 Phytophthora heveae

>AF266780 Phytophthora palmivora 


<span style="color: red;">The question says What are the species of each of the most likely sequences?, meaning I wanted you to see what where the sequences for each of the best hits, not all of them... (-10 pts) </span>

***

## Using HMMER to search for proteins with domains of interest

We are going to learn also how to use the information from databases to identify genes of interest that can serve us.

In this case, you are a Veterinarian Bioinformatitian. 

You have detected that a community of three animals (A chicken, *Gallus gallus*; a dog, *Canis familiaris*; and a cow, *Bos taurus*) that may have a disease associated with a mutation on the [Sonic The Hedgehog gene](https://medlineplus.gov/genetics/gene/shh/#resources).

You sequenced one copy of each the genes per species, and translated them intro proteins. 

However, due to an error in the machine, your sequences have been mixed with another hundred protein sequences from other species and genes in a file called `oh_this_is_a_mess.fasta`

A BLAST would be too much to run as you have a very small machine in your office, but you remember (thanks to your bioinformatics course ;) ) that you can run HMMER, a faster way of identifying protein sequences using Hidden Markov Models (HMM).

You also remember that [Pfam](http://pfam.xfam.org/) is a great database of protein domains that also contains HMM for these proteins.

Using your bioinformatics software, you discover that there is a reference gene for *Sonic The Hedgehog* in humans, in NCBI, in the accession number [NP_000184.1](http://www.ncbi.nlm.nih.gov/protein/NP_000184.1?report=fasta)

1. Copy the FASTA file and search for the domains in Pfam

2. What are the two Pfam-A matches you find?



***

> 1. Answer here: The two Pfam-A matches we found were Hint and HH_signal. 

***

3. Click on each of the domains. Go to the **Curation & model** link on the left.

4. Download the raw HMM for each domain. 

5. Log into the Cluster. Within the `Lab_4` folder create a `hmm_example` folder.

6. Upload the HMM files and the `oh_this_is_a_mess.fasta` files into the new folder

7. Merge both HMM files into a single file called `SHH.hmm`. Answer the following question:

***

> 2. How would you merge the two files? Add the code below.

Answer:  The code to merge the two files is:

cat Hint.hmm HH_signal.hmm > SHH.hmm

\![mergefile](sequences/merge.png)


***

8. Load the hmmer/3.3 module in the cluster using the following command:

```
module load hmmer/3.3
```
 
9. The following is a template to run HMMER searches:

```
hmmsearch --tblout output_table HMM_file query_sequences
```

Answer the following question:

***

> 3. Modify the template code by changing the `output_table`, `HMM_file` and `query_sequences` and add it below.

answer: hmmsearch --tblout search.txt SHH.hmm oh_this_is_a_mess.fasta 

<span style="color: red;"> Use code notation for code!(-1 pt) </span>

***

10. Run the modified code.

11. Check the `output_table` file. Could you find the sequences of interest? Explain the results below:

***

> 4. Answer here: The sequences of interest were found. Our XP_850450.1 sequences from hedgehog has 3 domains that satisfies threshold for both HH_signal and Hint. Similarly, XP_002687031.2 and NP_990152.1 also satisfies the threshold.

***

## Homework 2: 

How can you check if your candidate sequences are indeed the *Sonic The Hedgehog* genes?

***

> Answer here:
 Yes, the candidate sequences were indeed the *Sonic the Hedgehog* genes. From ncbi, we crated a database using HMMER. We obtained a reference protein of interest downloaded it.  The downloaded query protein sequence was used in PFAM to search for the domains. There were 2 domains present.Then, We downloaded the HMM model for both domains and merged both files. We ran the HMMER on the merged file. The sequence of interest were obtained in the output table, which revealed the sequence to be indeed *Sonic The Hedgehog* genes. We confirmed that the candidate sequence are indeed the Sonic hadgehog genes by looking at the description of target. The description of target mentioned 'sonic hedgehog protein' for the three anmials mentioned in answer 4 above. 


<span style="color: red;"> Excellent response! </span>

***



