---
title: "Lab 5: Homology and Phylogenetics (Section 1)"
author: Hannah Evers and Cammi MacKenzie
date: "10/14/2020"
output: html_document
---
<h2>
<span style="color: red;"> 50/50. Excellent job!!</span>
</h2>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Instructions:

- Section 1 will be done on Oct. 30
- Section 2 will be done on Nov. 6 
- Lab is due before Nov. 13


# Section 1: Homology and multiple sequence alignments

## Downloading sequences from Homologene

If you remember, homologous genes are genes that share common ancestry and can be functionally similar. 

There are many ways to obtain homologous genes. For example, running OrthoMCL/OrthoFinder in a set of genomes will identify all the genes of interest between the set of genomes.

In other cases, we can obtain genes of interest from online databases, such as the OrthoMCL DB or NCBI homologene.

Today, we will use NCBI Homologene. 

1. Go to this [link](https://www.ncbi.nlm.nih.gov/homologene/210)

2. Answer these questions:

***

> Questions:

> 1. What number of homologene is this? 210 
> 2. What gene/genic family is represented in this set of sequences? MutS; DNA repair protein
> 3. Based on the schematics on the right, which domains of the protein are conserved across the different species? MutS_1, MutS_2, and ABC_ATPase
> 4. Fill the following table using all the sequences from the homologene info

|Species|Common Name|Protein ID|Protein length|Number of domains found|
|---|---|---|---|---|
|*Homo sapiens*|Human|NP_000242.2|Three| (MutS_I, MutS_II, ABC_ATPase)| 
|*Canis lupus* | Dog | XP_538482.2 | Three | (MutS_I, MutS_II, ABC_ATPase)|
|*Bos taurus* | Cattle| NP_001029756.1 | Four | (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
| *Mus Musculus*| House Mouse | NP_032654.1 | Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
| *Rattus norvegicus* | Norway rat| NP_112320.1| Three | (MutS_I, MutS_II, ABC_ATPase)
| *Gallus gallus* | Chicken | XP_426110.4 | Four | (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
| *Xenopus tropicalis* | Tropical clawed frog| XP_002935427.2 |Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
|*Danio rerio*| zebrafish| NP_998689.1| Three| (MutS_I, MutS_II, ABC_ATPase)|
| *Drosophila melanogaster*| Fruit fly| NP_001246031.1 | Three| (MutS_I, MutS_II, MutSd)|
| *Anopheles gambiae* | PEST| XP_319476.4| Three| (MutS_I, MutS_II, MutS_V)|
| *Saccharomyces cerevisiae* | yeast| NP_014551.1| Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
| *Kluyveromyces lactis* | yeast| XP_455202.1| Three|(MutS_I, MutS_II, ABC_ATPase) |
| *Eremothecium gossypii* | Filamentous Fungus| NP_986150.1| Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
|*Schizosaccharomyces pombe*| Fission Yeast| XP_001713136.1| Three| (MutS_I, MutS_II, ABC_ATPase)|
|*Pyricularia oryzae*| Rice blast Fungus | XP_003718090.1| Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
|*Neurospora crassa* | Red bread mold fungus| XP_959643.1| Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|
|*Arabidopsis thaliana*| Thale cress| NP_566804.3| Four| (MutS_I, MutS_II, MutS_III, ABC_ATPase)|



***

3. Go to the Download links and download the mRNA and Protein sequences in FASTA format

4. Connect to the cluster and upload the sequences into a new folder inside of `MBB101` called `Lab_5`

5. Answer the following questions:

***

> Questions:

> 5. What is the number of sequences in each of the files uploaded? 17 and 17
> 6. Did you expect this number to be similar or different across the files? I expected them to be similar across the files because each mRNA sequence should have a corresponding protein sequence. 

***

## Multiple sequence alignment in MAFFT

Now, we will do some multiple sequence alignments using MAFFT.

If we remember from class, MAFFT is one of the most complete multiple sequence aligner programs that exist. 

It allows us to do many different kinds of searches, such as local alignments, global alignments, and iterative and progressive alignments focused on aligning highly conserved domains

6. Before you proceed with the alignment, go to the [MAFFT algortihms page](https://mafft.cbrc.jp/alignment/software/algorithms/algorithms.html) and answer the following questions based on the information on the algorithms MAFFT uses:


***
> Questions:

> 7. What kinds of algorithms from MAFFT (E-INS-i, L-INS-i or G-INS-i) would you use for the following datasets, and briefly explain your answers:
 - Proteins with multiple conserved domains: E-INS-i because we're looking across mutliple shared conserved domains.
 - Proteins with single conserved domains: L-INS-i because there is one larger shared conserved domain.
 - Proteins with a single short domain: G-INS-i because there is only a small shared conserved domain and we need to be able to look at the whole thing to be able to find that one shared region. 
 
***

7. Based on your answer above, align the protein sequence you downloaded using the algorithm of your choice in the cluster.

To do this, in your `Lab_5` do the following:

* Load the module by using the `module load mafft/7.471` command

* Use the following code for your alignment:
     * For L-INS-i
```
mafft --localpair --maxiterate 1000 input_file > output_file
```
     * For E-INS-i
```
mafft --genafpair --maxiterate 1000 input_file > output_file
mafft --genafpair --maxiterate 1000 homologene_prot.fasta > alin.fasta
```
     * For G-INS-i
```
mafft --globalpair --maxiterate 1000 input_file > output_file
```

Remember to change your `input_file` and `output_file` for the FASTA file and the output file (also in FASTA file).

8. Download your `output_file` into your local computer. Open the [NCBI MSA viewer page](http://msa.biojs.net/app/) on your browser and import your alignment using the Upload button and the Data file link.

9. Add the image of the MSA here:


***
> 8. \![MSA](protein_alignment[558..611].png)
***

10. Write a short summary of the MSA by playing with the options in the MSA aligner (Are there regions that are highly conserved?  Regions that are not that conserved? What are the species that look more similar to each other?) 

***
> 9. 218-224; 235-259; 377-382; 384-393; 559-556; 570-582; 586; 589; 536-537; 608; 619-622 look pretty conserved. 442 is 100% conserved. 811-826; 873-916 are pretty conserved as well.
598-603; 207-214; 395-390; 521-526; 801-810 look not conserved. 
The humans and the animals look most similar to each other while the yeast look most similar to each other. 
***

## Creating and searching a HMM model using our alignment

Now, we will search across a data set of FASTA proteins from the file we used last week but updated with some human proteins:

```
https://raw.githubusercontent.com/Tabima/MBB101/master/Lab_5/mess_with_human_proteins.fasta
```

11. Download the new FASTA file to your `Lab_4` folder. Count the number of sequences and answer the next question:

***
> 10. What is the number of sequences in the downloaded FASTA file? 101 

***

12. Now, we want to identify if there are proteins that match the MSH homologs we have in the new FASTA file. That means we will have to create a HMM profile of our MSH homologs and then search for the sequences!

13. To create the hmmer profile you need your alignment file of the MSH homologs in protein format, and the `hmmbuild` program.

The `hmmbuild` program creates a HMM model of your alignment (so, of your conserved domains!!) that you can use to search using `hmmsearch` how we did on last class!

To run it use the following command:

```
module load hmmer/3.3

hmmbuild MSH_homolog.hmm alignmnent_file.fasta
```

Answer the following question:

***
> 11. Summarize the process of building a HMM model

Gather your fasta protein files of your protein of interest from the various species you are looking at, align them into one file using the MAFFT algorithm of your choice, then you would build a HMMER to generate a HMM model that we could compare the unknown, jumbled sequences to.   

***


14. Finally, run `hmmsearch` and identify the protein with positive hits. Add them to the following table:

| Target Name  | Full Sequence E-value | Full Seq. Score | Best 1 domain E-value | Best 1 domain score |
|-------|-------|------|-------|--------|
|John_Cena_Protein| 0 | 1598.0 | 0 | 1597.8 |
     



