---
title: 'Lab 5: Homology and Phylogenetics (Section 2)'
author: Hannah Evers and Cammi MacKenzie
date: "11/13/2020"
output:  
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: inline
---
<h2>
<span style="color: red;"> 35/50. Y'all: When using code use code blocks! When using scientific names, use italics!!</span>
</h2>

# Instructions:

- Section 2 will be done on Nov. 13
- Lab is due before Nov. 20

## Creating a phylogenetic tree using protein data

One of the most common ways of creating phylogenetic trees is using protein data, as the length of the protein alignment is usually smaller than the nucleotide alignment. 

### Introduction to `IQ-tree`

In this lab, we will use one of the most recent programs to build phylogenetic trees: [`IQ-tree`](http://www.iqtree.org/).

`IQ-tree` uses a maximum likelihood approach and ultra-fast bootstrapping to reconstruct phylogenetic trees. 

`IQ-tree` is a very fast program, and it can use nucleotide, protein and other types of data to reconstruct phylogenies based on maximum likelihood.

In addition, if you remember from the theory lecture, `IQ-tree` also determines the best model for nucleotide/amino-acid substitution for the phylogenetic reconstruction, so pretty much `IQ-tree` can do it all.

## Running IQ-tree on protein data

So, we are going to use the protein alignment that we created last week.

To run IQ-tree, we need two super-simple commands:

1. Load the module for IQ-tree

```
module load iqtree
```

2. Use this magical command to run `IQ-tree` and automatically determine the best model of amino-acid evolution, run 1000 ultra-fast bootstraps and obtain the most likely tree.

```
iqtree2 -s your_protein_alignment -B 1000
```

3. Answer the following questions:

***
> 1. Check the output of the program as it runs. According to the output, what is the Best-fit model for our alignment? LG+I+G4

> 2. What are the best models according to Akaike information criteria, corrected Akaike information criteria, and Bayesian information criteria? Are they all different? Akaike: LG+F+I+G4, Corrected Akaike Information Criterion: LG+F+I+G4, and Bayesian Information Criterion: LG+I+G4. No, they are not all different. The Akaike models are the same while the Bayesian is the best-fit model. 

> 3. Use code to extract the Log-likelihood of the tree from the `.iqtree` file from the run. What is that value? How did you extract it? The Log-likelihood of the tree value is: -19731.4890. The code: grep  -w 'Log-likelihood' align.fasta.iqtree was used to extract that value. 

<span style="color: red;"> CODE NOTATION!! (-5 pts) </span>.

***

4. To view your final phylogenetic tree, download [FigTree](https://github.com/rambaut/figtree/releases) where the `dmg` is for MacOS, the `zip` for Windows, and the `tgz` for Linux. Install the program.

5. Download the `.treefile` file from the cluster to your computer. Open `FigTree` and open the downloaded `.treefile`.

6. You'll see a window like this. It asks you about what name you want to give the labels of the trees. I usually call them *bootstrap*:

7. Here's your tree! Now, all Maximum Likelihood trees are unrooted by definition. However, its easier to study the tree if you root it. In this case, you can root it on the node with the longest branch. To do so do `Trees -> Root tree -> Midpoint`.

8. You can also organize the tree for easy viewing by organizing it by early-emerging to late emerging. Use `Trees -> Order nodes -> Increasing` to do so.

9. Finally, observe the bootstrap values by selecting `Node Labels -> Bootstrap`. Good bootstrap support values are greater than 60%, highly supported values are greater than 75%. 

10. Answer the following question:

***
> 4. Summarize your findings (i.e.: Describe the relationships between the species; which nodes are well supported, which nodes are not; does this tree fit the expectation based on the taxonomy of the species used?)

Overall, it appears that all the animals are grouped together while all the yeast, fungi, and plants are grouped together in the lower half of the tree. You can see that the mouse and the rat are more closely related to each other which makes sense given their traits and overall structure. Then you have humans and dogs grouped together and then cattle being similar to both those group of the mouse, rat, human, and dog which makes sense as well as they all have four limbs, fingers/toes, hair (in some sense) etc. Then the chicken, the frog, and the zebrafish are all branching from each other away from the above five which makes sense as they are all animals as well but with differing features that have evolved overtime but are not close enough to be grouped in the same clade. To finish up the upper half the fruit fly and PEST branch off together which again makes sense due to the various tiny legs, wings, little bodies, etc. however still animals and having some relationship to the above animals. Then it branches from the very first ancestor to the branches of fungus and yeast and the one plant. You can see that the lactis yeast and the filamentous fungus are similar to each other which are similar to the bakers yeast. All of those together are then closely related to the rice blast fungus, red mold fungus, and fission yeast. The arabidopsis plant is then by itself in a clade which makes sense as the plant is more related to the fungi and yeast versus the animals but not that closely related, needing its own branch. Almost all of the nodes are well supported except for the nodes that connect the human and dog to the rat and mouse and the node connecting the fission yest to the rice blast and red mold fungi. The rat and mouse are always supported as well as the cattle to the human and dog, the zebrafish to the frog, the fruit fly and the PEST, bakers yeast to the filamentous fungus and the lactis yeast, that group to the other two fungi and yeast, and lastly the red mold and the rice blast fungi. There are a few 99 values in there as well which highly support the relationship found there. Overall, yes I believe the tree fits the expectation based on the taxonomies used as you can make sense as to why species are connected to each other and you can also understand and refute why some of the smaller bootstrap values are what they are. However, overall most of the bootstrap values are high and support the relationships found. 
***

## Running IQ-tree on nucelotide data

Now, lets do the same but using the `homologene_nucl.fasta` file you'll find in the moodle page. 

This file has **been previously aligned** using `MAFFT`, so this **SHOULD BE** the same file you got from the `mRNA` homologene file, and we will test this hypothesis. 

11. Align **YOUR** mRNA homologene file you obtained last week using `MAFFT`. Answer the following question.

***
> 5. Add the code for the MAFFT alignment of your mRNA file. 

mafft --genafpair --maxiterate 1000 homologene-2.txt > align_nucleotide.fasta

<span style="color: red;"> CODE NOTATION!! (-5 pts) </span>.

> 6. Why did you decide to use that algorithm? 

We are comparing multiple shared conserved domains just as we were when we were looking at the protein sequence. It's the same species and same domains as before. 

***

12. Run `IQ-tree` for each of the mRNA alignments. Fill the following table

| mRNA file | Best nucleotide model | Likelihood score of tree |
|---------|--------|--------|--------|----------|
| Your mRNA file (align_nucleotide.fasta) |   TIM3+F+I+G4    |    -46835.746    |        |          |    
| Professor's mRNA file (homologene_nucl.fasta) |   TIM3+F+R3     |    -49890.474     |        |          |    

13. Open both trees in FigTree. Answer the following question:

***
> 7. What is the most striking difference you see? 

There is a branch that in the homologene_nucl.fasta tree that is by itself and not included in the tree that we aligned. Its code is NM_113607.3. That is the major difference in the trees otherwise, the nodes and branching are similar. The bootstrap values are also different between the trees which is a result of the random sequence. 

<span style="color: red;"> Interesing. The weird sequence was supposed to be NM_008628.2</span>.

***

14. Open the professor's alignment in the [MSA viewer of NCBI](https://www.ncbi.nlm.nih.gov/projects/msaviewer/). After uploading the sequence, use the `Coloring -> Nucleic Acid Colors` option. Answer the following question:

***
> 8. There is one sequence that looks very weird (i.e. Colors don't match and there are long strands of insertions and deletions in the region). What sequence do you suspect is weird? 

The mus musculus (mouse) sequence. 

<span style="color: red;"> SCIENTIFIC NAME NOTATION!! (-5 pts) </span>.

***

15. The professor just noticed that he changed the sequence of *Mus musculus* (`gi|118130707|ref|NM_008628.2|`) and added the reverse strand instead of the forward strand... WHOOPS.

In this case, `MAFFT` can check all the sequences and correct the direction of the strand. Use the `--adjustdirectionaccurately` option of `MAFFT` to run it. 

So, if you used the `E-INS-i` algorithm, the syntax would be like this:

`mafft --adjustdirectionaccurately --genafpair --maxiterate 1000 homologene_nucl.fasta > professor_corrected.fasta`

Using this info, here's the homework:

## Homework: 

> Using the mis-alignment I sent, correct it using the `MAFFT` code above and run the tree again. Does the tree look like the mRNA you obtained? Write a paragraph about the comparison. 

Yes, now the tree looks like the mRNA that we obtained from our alignment. All of the sequences are the exact same and all of them are branching at the same nodes, lengths, and same bootstrap values. Again, we see that all the animals are grouped together while all the fungi and yeast are grouped together. Almost all the bootstrap values are significant, being 74% and higher. It shows the dog and cattle being more closely related followed by humans. The mouse and the rat are very similar to each other and not as closely related to the three above. Then, getting further out, the chicken, frog, and zebrafish are less closely related to all of those animals and then on a separate branch we have PEST and fruit fly very closely related as we approach the node that accounts for the most recent common ancestor of all the species in the tree. From there, it branches again into the section with all the fungi and yeast. The Lactis yeast and the Filamentous fungus are most closely related to each other and those are more closely related to the Bakers yeast which is more closely related to the Fission Yeast which are then related to the Rice blast fungus
and the Red bread mold which are most similar to each other. Then the Thale cress is in it's own branch as it is more closely related to the fungi and yeast than the animals but not as related to them as the other fungi and yeast species being analyzed. Most of these groupings and relatedness are the same as the tree we previously made (homologene) however, now we see the dog and cattle more related as the first tree grouped the dog and human more related but, these trees provided much better bootstrap values suggesting that the dog and cattle are most related rather than the dog and human.  

