---
title: 'Lab 5: Homology and Phylogenetics (Section 2)'
output: 
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: inline
---

# Instructions:

- Section 2 will be done on Nov. 13
- Lab is due before Nov. 20

## Creating a phylogenetic tree using protein data

One of the most common ways of creating phylogenetic trees is using protein data, as the length of the protein alignment is usually smaller than the nucleotide alignment. 

### Introduction to `IQ-tree`

In this lab, we will use one of the most recent programs to build phylogenetic trees: [`IQ-tree`](http://www.iqtree.org/).

`IQ-tree` uses a maximum likelihood approach and ultra-fast bootstrapping to reconstruct phylogenetic trees. 

`IQ-tree` is a very fast program, and it can use nucleotide, protein and other types of data to reconstruct phylogenies based on maximum likelihood.

In addition, if you remember from the theory lecture, `IQ-tree` also determines the best model for nucleotide/amino-acid substitution for the phylogenetic reconstruction, so pretty much `IQ-tree` can do it all.

## Running IQ-tree on protein data

So, we are going to use the protein alignment that we created last week.

To run IQ-tree, we need two super-simple commands:

1. Load the module for IQ-tree

```
module load iqtree
```

2. Use this magical command to run `IQ-tree` and automatically determine the best model of amino-acid evolution, run 1000 ultra-fast bootstraps and obtain the most likely tree.

```
iqtree2 -s your_protein_alignment -B 1000
```

3. Answer the following questions:

***
> 1. Check the output of the program as it runs. According to the output, what is the Best-fit model for our alignment?
LG+I+G4 chosen according to BIC is the best fit model.

> 2. What are the best models according to Akaike information criteria, corrected Akaike information criteria, and Bayesian information criteria? Are they all different?
* Akaike Information Criterion: LG+F+I+G4
* Corrected Akaike Information Criterion: LG+F+I+G4
* Bayesian Information Criterion: LG+I+G4

> 3. Use code to extract the Log-likelihood of the tree from the `.iqtree` file from the run. What is that value? How did you extracted it? 
* -19711.2154
* to find this I used the code less alin.fasta.iqtree and then /Log-likelihood

***


4. To view your final phylogenetic tree, download [FigTree](https://github.com/rambaut/figtree/releases) where the `dmg` is for MacOS, the `zip` for Windows, and the `tgz` for Linux. Install the program.

5. Download the `.treefile` file from the cluster to your computer. Open `FigTree` and open the downloaded `.treefile`.

6. You'll see a window like this. It asks you about what name you want to give the labels of the trees. I usually call them *bootstrap*:



7. Here's your tree! Now, all Maximum Likelihood trees are unrooted by definition. However, its easier to study the tree if you root it. In this case, you can root it on the node with the longest branch. To do so do `Trees -> Root tree -> Midpoint`.

8. You can also organize the tree for easy viewing by organizing it by early-emerging to late emerging. Use `Trees -> Order nodes -> Increasing` to do so.

9. Finally, observe the bootstrap values by selecting `Node Labels -> Bootstrap`. Good boostrap support values are greater than 60%, highly supported values are greater than 75%. 

10. Answer the following question:

***
> 4. Summarize your findings (i.e.: Describe the relationships between the species; which nodes are well supported, which nodes are not; does this tree fit the expectation based on the taxonomy of the species used?)

***

## Running IQ-tree on nucelotide data

Now, lets do the same but using the `homologene_nucl.fasta` file you'll find in the moodle page. 

This file has **been previously aligned** using `MAFFT`, so this **SHOULD BE** the same file you got from the `mRNA` homologene file, and we will test this hypothesis. 

11. Align **YOUR** mRNA homologene file you obtained last week using `MAFFT`. Answer the following question.

***
> 5. Add the code for the MAFFT alignment of your mRNA file. 
*mafft --localpair -maxiterate 1000 homologene_nucl.fasta > homologene_nucl_output.fasta

> 6. Why did you decide to use that algorithm?
*We decided we needed a local sequence alignment because were looking to compare the similarities within the file. We chose this type of allignment because we are comparing the homologs of the files.

***

12. Run `IQ-tree` for each of the mRNA alignments. Fill the following table

| mRNA file | Best nucleotide model | Likelihood score of tree |
|---------|--------|--------|--------|----------|
| Your mRNA file |TIM3+F+R3 chosen according to BIC|-50069.776| 
| Professor's mRNA file |TIM3+F+R3 chosen according to BIC|-49890.473|   

13. Open both trees in FigTree. Answer the following question:

***
> 7. What is the most striking difference you see? 

***

14. Open the professor's alignment in the [MSA viewer of NCBI](https://www.ncbi.nlm.nih.gov/projects/msaviewer/). After uploading the sequence, use the `Coloring -> Nucleic Acid Colors` option. Answer the following question:

***
> 8. There is one sequence that looks very weird (i.e. Colors don't match and there are long strands of insertions and deletions in the region). What sequence do you suspect its weird?

***

15. The professor just noticed that he changed the sequence of *Mus musculus* (`gi|118130707|ref|NM_008628.2|`) and added the reverse strand instead of the forward strand... WHOOPS.

In this case, `MAFFT` can check all the sequences and correct the direction of the strand. Use the `--adjustdirectionaccurately` option of `MAFFT` to run it. 

So, if you used the `E-INS-i` algorithm, the syntax would be like this:

`mafft --adjustdirectionaccurately --genafpair --maxiterate 1000 homologene_nucl.fasta > professor_corrected.fasta`

Using this info, here's the homework:

## Homework: 

> Using the mis-alignment I sent, correct it using the `MAFFT` code above and run the tree again. Does the tree look like the mRNA you obtained? Write a paragraph about the comparison.

