---
title: "Lab 5: Homology and Phylogenetics (Section 1)"
author: "Kelsey Joyce with Katie Crowley"
date: "10/14/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# Instructions:

- Section 1 will be done on Oct. 30
- Section 2 will be done on Nov. 6 
- Lab is due before Nov. 13


# Section 1: Homology and multiple sequence alignments

## Downloading sequences from Homologene

If you remember, homologous genes are genes that share common ancestry and can be functionally similar. 

There are many ways to obtain homologous genes. For example, running OrthoMCL/OrthoFinder in a set of genomes will identify all the genes of interest between the set of genomes.

In other cases, we can obtain genes of interest from online databases, such as the OrthoMCL DB or NCBI homologene.

Today, we will use NCBI Homologene. 

1. Go to this [link](https://www.ncbi.nlm.nih.gov/homologene/210)

2. Answer these questions:

***

> Questions:

> 1. What number of homologene is this?
* 210
> 2. What gene/genic family is represented in this set of sequences?
* Gene conserved in Eukaryota

<span style="color: red;"> The GENIC family represented is Msh2 (-5 pts) </span>.

> 3. Based on the schematics on the right, which domains of the protein are conserved across the different species?
* MUTSd (smart00533)
* MutS_V (pfam00488)
* MutS_I (pfam01624)
* MutS_III (pfam05192)
* MutS_II (pfam05188)
* MutS_I (pfam01624)
* MutS_II (pfam05188)
* ABC_ATPase (cl17201)
* MutS_III (cl17822)
> 4. Fill the following table using all the sequences from the homologene info

| Species                   | Common Name          | Protein ID     | Protein Length | Number of domains found                   |
|---------------------------|----------------------|----------------|----------------|-------------------------------------------|
| *Homo sapiens*              | Human                | NP_000242.2    | 934 aa         | 3 (MutS_I, MutS_II, ABC_ATPase            |
| *Canis lupus*               | Dog                  | XP_538482.2    | 934 aa         | 3 (MutS_I, MutS_II, ABC_ATPase)           |
| *Bos taurus*                | Cattle               | NP_001029756.1 | 934aa          | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Mus musculus*              | House mouse          | NP_032654.1    | 935aa          | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Rattus norvegicus*         | Norway rat           | NP_112320.1    | 933 aa         | 3 (MutS_I, MutS_II, ABC_ATPase)           |
| *Gallus gallus*             | Chicken              | NP_426110.4    | 968 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Xenopus tropicalis*        | Tropical clawed frog | XP_002935427.2 | 932 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Danio rerio*               | Zebrafish            | NP_998689.1    | 936 aa         | 3 (MutS_I, MutS_II, ABC_ATPase)           |
| *Drosophila melanogaster*   | Fruit fly            | NP_001246031.1 | 917 aa         | 3 (MutS_I, MutS_II, MUTSd)                |
| *Anopheles gambiae*         | Strain: PEST         | XP_319476.4    | 881 aa         | 3 (MutS_I, MutS_II, MutS_V)               |
| *Saccharomyces cerevisiae*  | Strain: S288C        | NP_014551.1    | 964 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Kluyveromyces lactis*      | Strain: NRRL Y-1140  | XP_455202.1    | 956 aa         | 3 (MutS_I, MutS_II, ABC_ATPase)           |
| *Eremothecium gossypii*     | Strain: ATCC 10895   | NP_986150.1    | 956 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Schizosaccharomyces pombe* | Fission yeast        | XP_001713136.1 | 982 aa         | 3 (MutS_I, MutS_II, ABC_ATPase)           |
| *Pyricularia oryzae*        | Strain: 70-15        | XP_003718090.1 | 924 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Neurospora crassa*         | Strain: OR74A        | XP_959643.1    | 937 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |
| *Arabidopsis thaliana*      | Thale cress          | NP_566804.3    | 937 aa         | 4 (MutS_I, MutS_II, MutS_III, ABC_ATPase) |



***

3. Go to the Download links and download the mRNA and Protein sequences in FASTA format

4. Connect to the cluster and upload the sequences into a new folder inside of `MBB101` called `Lab_5`

5. Answer the following questions:

***

> Questions:

> 5. What is the number of sequences in each of the files uploaded? 
* 17 sequences
> 6. Did you expect this number to be similar or different across the files?
* Yes this was expected because we are looking at the same species with the same data we are just using different file formats

***

## Multiple sequence alignment in MAFFT

Now, we will do some multiple sequence alignments using MAFFT.

If we remember from class, MAFFT is one of the most complete multiple sequence aligner programs that exist. 

It allows us to do many different kinds of searches, such as local alignments, global alignments, and iterative and progressive alignments focused on aligning highly conserved domains

6. Before you proceed with the alignment, go to the [MAFFT algortihms page](https://mafft.cbrc.jp/alignment/software/algorithms/algorithms.html) and answer the follwing questions based on the information on the algorithms MAFFT uses:


***
> Questions:

> 7. What kinds of algorithms from MAFFT (E-INS-i, L-INS-i or G-INS-i) would you use for the following datasets, and briefly explain your answers:
 - Proteins with multiple conserved domains: I would use the E-INS-i algorithm for this. This is because you can use pairwise alignment to compare domains. Then domains taht are shared tell us about the evolutionary history of the organisms. 
 - Proteins with single conserved domains: I would use the L-INS-i for this because the L-INS-i can allign a set of sequence containing sequences around one alignable domain. 
 - Proteins with a single short domain: For this I would use the G-INS-i algorithm. This is because it uses the Needleman-Wunsch algorithm to compare regions. This is why it it is idea for the single short domain.
 
***

7. Based on your answer above, align the protein sequence you downloaded using the algorithm of your choice in the cluster.

To do this, in your `Lab_5` do the following:

* Load the module by using the `module load mafft/7.471` command

* Use the following code for your alignment:
     * For L-INS-i
```
mafft --localpair --maxiterate 1000 input_file > output_file
```
     * For E-INS-i
```
mafft --genafpair --maxiterate 1000 input_file > output_file
mafft --genafpair --maxiterate 1000 homologene_prot.fasta > alin.fasta
```
     * For G-INS-i
```
mafft --globalpair --maxiterate 1000 input_file > output_file
```

Remember to change your `input_file` and `output_file` for the FASTA file and the output file (also in FASTA file).

8. Download your `output_file` into your local computer. Open the [NCBI MSA viewer page](http://msa.biojs.net/app/) on your browser and import your alignment using the Upload button and the Data file link.

9. Add the image of the MSA here:

***
> 8. 

!["MSA"](Align.jpeg)

***

10. Write a short summary of the MSA by playing with the options in the MSA aligner (Are there regions that are highly conserved?  Regions that are not that conserved? What are the species that look more similar to each other?)

***
> 9. 
* Looking at organisms that have similar characteristics (i.e. the organisms with vertebrates) many similarities in the sequences can be seen. These large areas of similar sequences are likely to be conserved areas. Areas that have large gaps are likely to be regions that are not conserved. 

***

## Creating and searching a HMM model using our alignment

Now, we will search across a data set of FASTA proteins from the file we used last week but updated with some human proteins:

```
https://raw.githubusercontent.com/Tabima/MBB101/master/Lab_5/mess_with_human_proteins.fasta
```

11. Download the new FASTA file to your `Lab_4` folder. Count the number of sequences and answer the next question:

***
> 10. What is the number of sequences in the downloaded FASTA file?
* 101 sequences

***

12. Now, we want to identify if there are proteins that match the MSH homologs we have in the new FASTA file. That means we will have to create a HMM profile of our MSH homologs and then search for the sequences!

13. To create the hmmer profile you need your alignment file of the MSH homologs in protein format, and the `hmmbuild` program.

The `hmmbuild` program creates a HMM model of your alignment (so, of your conserved domains!!) that you can use to search using `hmmsearch` how we did on last class!

To run it use the following command:

```
module load hmmer/3.3

hmmbuild MSH_homolog.hmm alignmnent_file.fasta
```

Answer the following question:

***
> 11. Summarize the process of building a HMM model
* Download sequences, align to show similarities and look for regions that are shared. Hmmr is to look for these similarities. 

***


14. Finally, run `hmmsearch` and identify the protein with positive hits. Add them to the following table:

| Target name       | Full Sequence E-value | Full Seq. Score | Best 1 domain E-value | Best 1 domain score |
|-------------------|-----------------------|-----------------|-----------------------|---------------------|
| John_cena_protein | 0                     | 1597.9          | 0                     | 1597.7              |
| XP_012783992.1    | 0                     | 1392.7          | 2e-281                | 928.5               |    



