---
title: 'Lab 5: Homology and Phylogenetics (Section 2)'
author: "Ryan Swanke and Matthew Finnegan"
output:
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: inline
---

<h2>
<span style="color: red;"> 45/50. Use scientific notation for species names. </span>
</h2>

# Instructions:

- Section 2 will be done on Nov. 13
- Lab is due before Nov. 20

## Creating a phylogenetic tree using protein data

One of the most common ways of creating phylogenetic trees is using protein data, as the length of the protein alignment is usually smaller than the nucleotide alignment. 

### Introduction to `IQ-tree`

In this lab, we will use one of the most recent programs to build phylogenetic trees: [`IQ-tree`](http://www.iqtree.org/).

`IQ-tree` uses a maximum likelihood approach and ultra-fast bootstrapping to reconstruct phylogenetic trees. 

`IQ-tree` is a very fast program, and it can use nucleotide, protein and other types of data to reconstruct phylogenies based on maximum likelihood.

In addition, if you remember from the theory lecture, `IQ-tree` also determines the best model for nucleotide/amino-acid substitution for the phylogenetic reconstruction, so pretty much `IQ-tree` can do it all.

## Running IQ-tree on protein data

So, we are going to use the protein alignment that we created last week.

To run IQ-tree, we need two super-simple commands:

1. Load the module for IQ-tree

```
module load iqtree
```

2. Use this magical command to run `IQ-tree` and automatically determine the best model of amino-acid evolution, run 1000 ultra-fast bootstraps and obtain the most likely tree.

```
iqtree2 -s your_protein_alignment -B 1000
```

3. Answer the following questions:

***
> 1. Check the output of the program as it runs. According to the output, what is the Best-fit model for our alignment?

LG+I+G4

> 2. What are the best models according to Akaike information criteria, corrected Akaike information criteria, and Bayesian information criteria? Are they all different?

Akaike info criteria: LG+F+I+G4

Corrected Akaike info criteria: LG+F+I+G4

Bayesian info creiteria: LG+I+G4

the akaike ones 

> 3. Use code to extract the Log-likelihood of the tree from the `.iqtree` file from the run. What is that value? How did you extract it? 

the value is -19731.4890

we used `cat prot_results.fasta.iqtree` and looked by hand

***

4. To view your final phylogenetic tree, download [FigTree](https://github.com/rambaut/figtree/releases) where the `dmg` is for MacOS, the `zip` for Windows, and the `tgz` for Linux. Install the program.

5. Download the `.treefile` file from the cluster to your computer. Open `FigTree` and open the downloaded `.treefile`.

6. You'll see a window like this. It asks you about what name you want to give the labels of the trees. I usually call them *bootstrap*:

\![](Lab_5/img1.png)

7. Here's your tree! Now, all Maximum Likelihood trees are unrooted by definition. However, its easier to study the tree if you root it. In this case, you can root it on the node with the longest branch. To do so do `Trees -> Root tree -> Midpoint`.

8. You can also organize the tree for easy viewing by organizing it by early-emerging to late emerging. Use `Trees -> Order nodes -> Increasing` to do so.

9. Finally, observe the bootstrap values by selecting `Node Labels -> Bootstrap`. Good boostrap support values are greater than 60%, highly supported values are greater than 75%. 

10. Answer the following question:

***
> 4. Summarize your findings (i.e.: Describe the relationships between the species; which nodes are well supported, which nodes are not; does this tree fit the expectation based on the taxonomy of the species used?)

all relationships appeared to match up very well among what one would expect based on taxonomy. all mammals shared a monophyletic group, all animals shared a monophyletic group, both insects shared a monophyletic group, all yeasts shared a monophyletic group, all fungi shared a monophyletic group, and all non-animals shared a monophyletic group.

almost all nodes were above a 75 bootstrap value. the one between the human/dog group and mouse/rat was 73, and the one between XP_003718090.1/XP_959643.1 and XP>001713136.1 was 47

***

## Running IQ-tree on nucelotide data

Now, lets do the same but using the `homologene_nucl.fasta` file you'll find in the moodle page. 

This file has **been previously aligned** using `MAFFT`, so this **SHOULD BE** the same file you got from the `mRNA` homologene file, and we will test this hypothesis. 

11. Align **YOUR** mRNA homologene file you obtained last week using `MAFFT`. Answer the following question.

***
> 5. Add the code for the MAFFT alignment of your mRNA file. 

`mafft --genafpair --maxiterate 1000 homologene_mrna.fasta > homologene_mrna_mafft.fasta`


> 6. Why did you decide to use that algorithm?

we're using this code because there are multiple conserved domains that we are comparing


***

12. Run `IQ-tree` for each of the mRNA alignments. Fill the following table

| mRNA file | Best nucleotide model | Likelihood score of tree |
|---------|--------|--------|--------|----------|
| Your mRNA file |    TIM3+F+I+G4    |  -46836.466      |        |          |    
| Professor's mRNA file |     TIM3+F+R3   |  -49890.473      |        |          |    

13. Open both trees in FigTree. Answer the following question:

***
> 7. What is the most striking difference you see?

the mus musculus is completely separate from the tree

<span style="color: red;"> SCIENTIFIC NOTATION FOR SPECIES NAMES! (-5 pts)</span>


***

14. Open the professor's alignment in the [MSA viewer of NCBI](https://www.ncbi.nlm.nih.gov/projects/msaviewer/). After uploading the sequence, use the `Coloring -> Nucleic Acid Colors` option. Answer the following question:

***
> 8. There is one sequence that looks very weird (i.e. Colors don't match and there are long strands of insertions and deletions in the region). What sequence do you suspect its weird?

the *Mus musculus*'s sequence is very different

***

15. The professor just noticed that he changed the sequence of *Mus musculus* (`gi|118130707|ref|NM_008628.2|`) and added the reverse strand instead of the forward strand... WHOOPS.

In this case, `MAFFT` can check all the sequences and correct the direction of the strand. Use the `--adjustdirectionaccurately` option of `MAFFT` to run it. 

So, if you used the `E-INS-i` algorithm, the syntax would be like this:

`mafft --adjustdirectionaccurately --genafpair --maxiterate 1000 homologene_nucl.fasta > professor_corrected.fasta`

Using this info, here's the homework:

## Homework: 

> Using the mis-alignment I sent, correct it using the `MAFFT` code above and run the tree again. Does the tree look like the mRNA you obtained? Write a paragraph about the comparison.

the corrected file and the file from the data we found were extremely close. the node positions were all exactly the same, with the bootstrap values varying slightly if at all. the largest variance between bootstrap values was 5.
